### PR TITLE
Enable matrix orderings

### PR DESCRIPTION
**Pull request by joschmitt**
I tried to use matrix orderings in Oscar, but ran into problems.
With these changes, it works for me, but I don't know anything about this code. Maybe somebody can have a look?

### CODE REVIEW

**Comment by fieker** [comment]
better than my "fix". What would be even better is to add the exampel to test (which I did not do)

**Comment by joschmitt** [comment]
Oh, sorry, I completely missed your PR.

**Comment by tthsqe12** [comment]
I think this pr is better. Can we have a test?

**Comment by joschmitt** [comment]
Yes, one moment.

**Comment by joschmitt** [comment]
So I wanted to edit `test/Rings/groebner-test.jl` for this, but realized that this file does not get included anywhere. Is this intended?

**Comment by fieker** [comment]
On Mon, Mar 14, 2022 at 07:08:22AM -0700, Johannes Schmitt wrote:
> So I wanted to edit `test/Rings/groebner-test.jl` for this, but realized that this file does not get included anywhere. Is this intended?
> 
I think this is a hangover from the fact that gb/ Q, Qt is currently
really slow (modstd disabled)...
I'd add the file, disable all but your test

**Comment by joschmitt** [comment]
Let's wait for #1161 here. I will rebase once that PR is merged.

File: test/Rings/groebner-test.jl [hunk 1]
```diff
@@ -1,37 +1,39 @@
-@testset "groebner" begin
-    R, (x, y) = PolynomialRing(QQ, ["x", "y"])
-    I = ideal(R,[x*y^2 - x, x^3 - 2*y^5])
-    @test leading_ideal(I) == ideal(R,[x*y^2, x^4, y^5])
-    @test leading_ideal(I, :lex) == ideal(R,[y^7, x*y^2, x^3])
-end
-
-@testset "groebner leading ideal" begin
-   R, (t, x, y, z) = PolynomialRing(QQ, ["t", "x", "y", "z"])
+#@testset "groebner" begin
```
Comment (fingolfin) [inline test/Rings/groebner-test.jl]: Please add a comment explaining why this is disabled.

Also, perhaps use a multiline comment (`#= ... =#`) for a smaller diff?

Basically, code should either be deleted; or if it is retained and for some reason just commented out, this should be explained in a comment.

Comment (joschmitt) [reply test/Rings/groebner-test.jl]: This is outdated anyway; I'm waiting for #1161 .
But I will do so in the future.

**Comment by joschmitt** [comment]
https://github.com/oscar-system/Oscar.jl/pull/1160/commits/01f2febed2469b2d876463af21d09887e3c5ca56 adds a flag to `groebner_basis` to skip the check whether the ordering is global. Since this is nothing one should try at home, the flag is deliberately long (`enforce_global_ordering`) and a keyword argument.
I hope this is fine for everybody. If you don't like it, then I do not insist on it.

**Review by tthsqe12 — APPROVED** [review]
Looks good from here.

**Pull request merged**